### PR TITLE
SONARJAVA-770 Support annotations at start and end of modifiers

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java [hunk 1]
```diff
@@ -65,23 +66,45 @@ public void visitNode(Tree tree) {
...
+      }
+    }
+    return testOnlyAnnotationsAreLeft(modifiersIterator);
+  }
+
+  /**
+   * Move iterator on the first element which is not an annotation
+   */
+  private static void skipAnnotations(ListIterator<ModifierTree> modifiersIterator) {
+    while (modifiersIterator.hasNext() && modifiersIterator.next().is(Kind.ANNOTATION)) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java]: MAJOR Either remove or fill this block of code. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00108)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java [hunk 2]
```diff
@@ -65,23 +66,45 @@ public void visitNode(Tree tree) {
   }
 
   private static ModifierTree getFirstBadlyOrdered(ModifiersTree modifiersTree) {
-    int modifierIndex = -1;
+    ListIterator<ModifierTree> modifiersIterator = modifiersTree.listIterator();
+    skipAnnotations(modifiersIterator);
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java]: It should be way simpler to generate a new list of `modifierTree` from the provided list, but without all the annotations at its start and at its end of the list. and then use the same logic as before.

Comment (saberduck) [inline java-checks/src/main/java/org/sonar/java/checks/ModifiersOrderCheck.java]: As discussed, this is in fact equivalent and having it in the same place allows to simplify previous logic a bit.

**Pull request merged**